ci/test/03_test_script.sh

```diff
@@ -132,6 +132,44 @@ bash -c "${PRINT_CCACHE_STATISTICS}"
 du -sh "${DEPENDS_DIR}"/*/
 du -sh "${PREVIOUS_RELEASES_DIR}"
 
+if [[ ${BARE_METAL_RISCV} == "true" ]]; then
+    cmake --build . --target bitcoin_consensus
+
+    echo -e "#include <script/script_error.h>\n#include <cassert>\n extern \"C\" void _start() { assert(ScriptErrorString(ScriptError_t::SCRIPT_ERR_UNKNOWN_ERROR).size() > 0);}" > test.cpp
+
+    /opt/riscv-ilp32/bin/riscv32-unknown-elf-g++ -I "${BASE_ROOT_DIR}"/src -g -std=c++20 -march=rv32gc -mabi=ilp32 -c test.cpp -o test.o
+
+    echo -e "#include <sys/stat.h>
+          void _exit(int code) { while(1); }
+          int _sbrk(int incr) { return 0; }
+          int _write(int file, char *ptr, int len) { return 0; }
+          int _close(int file) { return -1; }
+          int _fstat(int file, struct stat *st) { st->st_mode = S_IFCHR; return 0; }
+          int _isatty(int file) { return 1; }
+          int _lseek(int file, int ptr, int dir) { return 0; }
+          int _read(int file, char *ptr, int len) { return 0; }
+          int _kill(int pid, int sig) { return -1; }
+          int _getpid(void) { return -1; }" > syscalls.c
+
+    /opt/riscv-ilp32/bin/riscv32-unknown-elf-gcc -g -march=rv32i -mabi=ilp32 -c syscalls.c -o syscalls.o
+
+    /opt/riscv-ilp32/bin/riscv32-unknown-elf-g++ -g -std=c++20 -march=rv32gc -mabi=ilp32 \
+        -static-pie -static-libgcc -static-libstdc++ -nostartfiles -nodefaultlibs -nostdlib \
+        /opt/riscv-ilp32/lib/gcc/riscv32-unknown-elf/14.2.0/crtbegin.o \
+        test.o \
+        syscalls.o \
+        -Wl,--whole-archive \
+        src/libbitcoin_consensus.a \
+        -Wl,--no-whole-archive \
+        src/crypto/libbitcoin_crypto.a \
+        src/secp256k1/lib/libsecp256k1.a \
+        /opt/riscv-ilp32/riscv32-unknown-elf/lib/libstdc++.a \
+        /riscv/newlib/build/riscv32-unknown-elf/newlib/libc.a \
+        /riscv/newlib/build/riscv32-unknown-elf/newlib/libm.a \
```

Comment:
Any specific reason to use the intermediate build and not the installed libraries from `/opt/newlib`, here?